example/simple_filter.cpp

```diff
@@ -0,0 +1,38 @@
+//include libraries for image io, kernel definition and convolution
+#include <boost/gil/image.hpp>
+#include <boost/gil/typedefs.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+
+int main(int argc,char *argv[]) {
+    using namespace boost::gil;
+    if(argc!=2){
+        printf("Usage: simple_filter <image_path>");
+        return 0;
+    }
+    rgb8_image_t img;
+    //read image
+    read_image(*argv[1], img, jpeg_tag{});
+
+    // Convolve the rows and the columns of the image with a fixed kernel
+    gray8_image_t convolved(img.dimensions());
+    gray8_image_t convolved2(img.dimensions());
+
+    // difference kernel
+    float sharpen_1[]={-0.05f,0.0f,0.05f};
+    
+
+    // making resizable kernel object
+    kernel_1d<float> kernel2(sharpen_1,3,1);
+    //compute horizontal edges
```

Comment:
The comment seems to be repeating what the line of C++ code shows.
It rather should explain purpose of computing those edges.

Imagine I know very little about image processing and I read your code for the first time, do I have a chance to learn what algorithm your code implements? 

To me, you should give a comment explaining the algorithm, what it does, what are the steps.